tests/core/test_datamodules.py

```diff
@@ -82,39 +80,35 @@ def test_can_prepare_data(local_rank, node_rank):
     dm._has_prepared_data = False
     node_rank.return_value = 1
     local_rank.return_value = 0
-    assert not trainer.data_connector.can_prepare_data()
 
     trainer.data_connector.prepare_data()
     assert dm.random_full is None
 
     node_rank.return_value = 0
     local_rank.return_value = 1
-    assert not trainer.data_connector.can_prepare_data()
 
     trainer.data_connector.prepare_data()
     assert dm.random_full is None
 
     # 2 dm
     # prepar per node = True
     # local rank = 0 (True)
-    trainer.prepare_data_per_node = True
+    dm.prepare_data_per_node = True
+    trainer.lightning_module.prepare_data_per_node = True
     local_rank.return_value = 0
 
     # is_overridden prepare data = True
     # has been called
     # False
     dm._has_prepared_data = True
-    assert not trainer.data_connector.can_prepare_data()
```

Comment:
by removing these asserts, this test now becomes a no-op. Will we test it in a different way?

Reply:
good catch. sure, I will add test to see if` datamodule.prepare_data` has been called or not.